example_parlai_internal/README.md

```diff
@@ -0,0 +1,31 @@
+# Internal subfolder
+
+You can set up your own custom agents and tasks, and even create a
+separate git repository here to manage your models without the risk of accidentally pushing them to the public ParlAI repo.
+
+Start by creating a new folder named parlai_internal and copying the contents of this folder.
+ (We've added this to .gitignore already.)
+
+```bash
+cd ~/ParlAI
+mkdir parlai_internal
+cp -r example_parlai_internal/* parlai_internal
+cd parlai_internal
+```
+
+We've ignored this folder, but that's it. If you want to set this up as
+a separate git repository (e.g. for version control) you can follow the standard steps
+for creating a new repo (feel free to do this however you prefer).
+
+```bash
+git init
+git add .
+git commit -m "Initialize parlai_internal"
+```
+
+You can connect this to a new github repository if desired.
+[Create a new repo](https://github.com/new) (you don't need to initialize
+with a README), and then follow the instructions to push
+an existing repository from command line.
+
+Now you can do `from parlai_internal.X.Y import Z` to use your custom modules.
```

Comment:
It also provides a number of shortcuts which mirror the public repo. For example, you can run:

```bash
python examples/train_model -m internal:secret -t convai2
```

This will automatically look for code in `parlai_internal.agents.secret.SecretAgent`.

-----
also an ditto for tasks

Reply:
Also for model files, you can add an internal zoo! See PR #1735. Can you add documentaiton for how to access models in your internal zoo?

Reply:
Also please add a short section to "Tips n Tricks" that says you can do this.

Reply:
I'm actually wondering if all of this content belongs in a README or if we should follow the same philosophy of putting this in docs instead.

Reply:
I suggest adding the text from the beginning of this thread ("It also provides a number of shortcuts...") to this document. Then in tips and tricks, add a brief description and a link to your README.